indexing-service/src/main/java/io/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -689,6 +706,28 @@ public Response doGetLog(
     }
   }
 
+  @GET
+  @Path("/dataSources/{dataSource}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getRunningTasksByDataSource(@PathParam("dataSource") String dataSource,
+      @Context HttpServletRequest request)
+  {
+    List<TaskRunnerWorkItem> taskRunnerWorkItemList = new ArrayList<>();
+    Optional<TaskRunner> ts = taskMaster.getTaskRunner();
+    if (ts.isPresent()) {
+      Collection<? extends TaskRunnerWorkItem> runningTasks = ts.get().getRunningTasks();
+      if (runningTasks == null || runningTasks.isEmpty()) {
+        return Response.ok("No running tasks found").build();
+      }
+      for (TaskRunnerWorkItem taskRunnerWorkItem : runningTasks) {
```

Comment:
this can be made more functional similar to:

```java
runningTasks.filter(t -> dataSource.equals(t.getDataSource())).collect(Collectors.toList())
```

Reply:
Thanks. Fixed